tests/Support/DataClassTest.php

```diff
@@ -120,3 +136,18 @@ public static function fromDummyModel(DummyModel $model)
         return new self($model->id);
     }
 }
+
+#[MapName(SnakeCaseMapper::class)]
+#[WithTransformer(DateTimeInterfaceTransformer::class, 'd-m-Y')]
+#[WithCast(DateTimeInterfaceCast::class, format: 'Y-m-d')]
+class ParentData extends Data
+{
+}
+
+class ChildData extends ParentData
+{
+    public function __construct(
+        public DateTimeInterface $dateTime
+    ) {
+    }
+}
```

Comment:
Can you move this inside the test and give it more test specific names like TestRecursiveAttributesChildData?